mautic/mautic
# DPMMA-375 Improved warnings for unpublished emails in campaign

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ N ]
| New feature/enhancement? (use the a.x branch)      | [ Y ]
| Deprecations?                          | [ Y ]
| BC breaks? (use the c.x branch)        | [ N ]
| Automated tests included?              | [ Y ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
This PR changes the flash messages for unpublished emails in campaigns:
- Checks for publishUp and publishDown dates (previously only isPublished flag was checked)
- The message type is changed to warning - it will have the yellow border
- Added a link to unpublished email in the message
- Warning for each unpublished email instead of one general message

![image](https://github.com/mautic/mautic/assets/8580942/cbfef061-06aa-4efa-87cc-ec645e32481a)




#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Add unpublished email to campaign
3. Add published email with publishUp and publishDown from the past
4. Test that the warnings are displayed correctly
5. Warnings should not prevent the campaign from being saved


#### Deprecations:
- `Mautic\CampaignBundle\Tests\Service\Campaign` to be removed in 6.0; use CampaignAuditService instead. Class name can be confused with Campaign entity, can't import it without alias or reference to a namespace.
- `Mautic\CampaignBundle\Tests\Service\CampaignTest` to be removed in 6.0
- `Mautic\EmailBundle\Entity\EmailRepository::isOneUnpublished` to be removed in 6.0 with no replacement
